### PR TITLE
Add new Amazon SQS module

### PR DESCRIPTION
**Pull request by loia**
- New module for Amazon Simple Queue Service (SQS).
- Can be used to create, update and delete SQS queues.

### CODE REVIEW

**Comment by sivel** [comment]
At a minimum, the file needs to have a `.py` file extension.  I've not done any further review.

**Comment by loia** [comment]
Added the `.py` file extension.

**Comment by wimnat** [comment]
This is similar to sqs_queue module i wrote (https://github.com/wimnat/ansible-modules/blob/master/sqs_queue/sqs_queue.py)

I would rename this module to sqs_queue.py to differentiate it from anyone writing a module for writing messages to an sqs queue e.g a module called sqs_message.py

This module has lots of queue attributes but the boto docs state that only VisibilityTimeout is valid at this time - http://boto.readthedocs.org/en/latest/ref/sqs.html#boto.sqs.queue.Queue.set_attribute.  Are the boto docs incorrect?

**Comment by loia** [comment]
@wimnat Thanks for the feedback.

As per your suggestion, I've renamed the module to sqs_queue.py

Yes the boto docs appear to be inaccurate as I've tested each attribute using boto version 2.33.0 and they are working.  Maybe it was fixed in boto or the AWS API and the docs just weren't updated.

**Comment by gregdek** [comment]
Thanks for the changes @loia. @wimnat can you re-review?

File: cloud/amazon/sqs_queue.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  - Create or delete AWS SQS queues.
+  - Update attributes on existing queues.
+author: Alan Loi (@loia)
+requirements:
+  - "boto >= 2.33.0"
+options:
+  state:
+    description:
+      - Create or delete the queue
+    required: true
```
Comment (wimnat) [inline cloud/amazon/sqs_queue.py]: State not required if you're setting a default

File: cloud/amazon/sqs_queue.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        message_retention_period=dict(type='int'),
+        maximum_message_size=dict(type='int'),
+        delivery_delay=dict(type='int'),
+        receive_message_wait_time=dict(type='int'),
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True)
+
```
Comment (wimnat) [inline cloud/amazon/sqs_queue.py:212]: You need to add a handler for HAS_BOTO
E.g.
    if not HAS_BOTO:
        module.fail_json(msg='boto required for this module')

File: cloud/amazon/sqs_queue.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module)
+    connection = boto.sqs.connect_to_region(region)
+
```
Comment (wimnat) [inline cloud/amazon/sqs_queue.py:225]: Handle no region and use connect_to_aws method to support boto profiles.
E.g.
    if region:
        try:
            connection = connect_to_aws(boto.sqs, region, **aws_connect_params)
        except (boto.exception.NoAuthHandlerFound, StandardError), e:
            module.fail_json(msg=str(e))
    else:
        module.fail_json(msg="region must be specified")

**Comment by loia** [comment]
@wimnat I've addressed all of your comments:
- docfix - `state` option not required
- check `boto` library installed
- check AWS `region` option and credentials

**Comment by gregdek** [comment]
Thx @loia -- @wimnat back to you :)

**Comment by loia** [comment]
Added some docfixes - version_added and missing default values.

**Comment by loia** [comment]
@wimnat Did you have any further comments on this PR?

**Comment by wimnat** [comment]
+1

**Comment by gregdek** [comment]
Thanks for the review @wimnat and thanks @loia for the new module. 

Marking for inclusion.

**Pull request merged**